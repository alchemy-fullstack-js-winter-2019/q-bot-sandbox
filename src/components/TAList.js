import React from 'react';
import { useFirestore } from './useFirestore';

export default function TAList() {
  return (
    <h1>TA List</h1>
  );
}
