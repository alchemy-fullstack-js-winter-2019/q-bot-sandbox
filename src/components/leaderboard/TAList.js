import React from 'react';
import { useFirestore } from '../connectFirestore';
import { taCollection } from '../../services/firebase';
import TA from './TA';

export default function TAList({ handleDelete }) {
  const question = useFirestore(taCollection, [])
  const list = question.map(doc => {
    return <TA handleDelete={handleDelete} key={doc.id} id={doc.id} name={doc.name} cohort={doc.cohort} />
  })
  return (
    <>
      <h1>TA Leader Board</h1>
      <table className={'leaderTable'}>
        <thead>
          <tr>
            <th className={'tableHeader'}>Name</th>
            <th className={'tableHeader'}>Cohort</th>
            <th className={'tableHeader'}>Claimed Tickets</th>
            <th className={'tableHeader'}>Delete</th>
          </tr>
        </thead>
        <tbody>
          {list}
        </tbody>
      </table>
    </>
  );
}
