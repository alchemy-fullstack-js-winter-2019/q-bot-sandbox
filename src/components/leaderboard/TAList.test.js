import React from 'react';
import { shallow } from 'enzyme';
import TAList from './TAList';

describe('TAList', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <TAList handleDelete={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
