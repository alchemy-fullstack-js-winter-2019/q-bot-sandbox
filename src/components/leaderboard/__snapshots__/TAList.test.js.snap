// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TAList matches a snapshot 1`] = `
<Fragment>
  <h1>
    TA Leader Board
  </h1>
  <table
    className="leaderTable"
  >
    <thead>
      <tr>
        <th
          className="tableHeader"
        >
          Name
        </th>
        <th
          className="tableHeader"
        >
          Cohort
        </th>
        <th
          className="tableHeader"
        >
          Claimed Tickets
        </th>
        <th
          className="tableHeader"
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</Fragment>
`;
